mlflow/models/evaluation/base.py

```diff
@@ -413,6 +413,11 @@ def _hash_array_like_element_as_bytes(v):
 
         data = data.applymap(_hash_array_like_element_as_bytes)
         return _hash_uint64_ndarray_as_bytes(pd.util.hash_pandas_object(data))
+    elif isinstance(data, np.ndarray) and len(data) > 0 and isinstance(data[0], list):
+        # convert numpy array of lists into numpy array of numpy arrays
+        # because lists are not hashable
+        hashable = np.array(json.dumps(val) for val in data)
```

Comment:
instead of `json.dumps(val)`, can we do `str(val)`? This will handle the case where the data isn't JSON-serializable, e.g. you're dealing with a list of objects